tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/core/types.h

```diff
@@ -45,6 +46,8 @@ struct PlannerResponse
 {
   tesseract_common::JointTrajectory joint_trajectory; /**< @brief The joint trajectory */
   tesseract_common::StatusCode status;                /**< @brief The status information */
+  std::vector<Waypoint::Ptr> succeeded_waypoints;     /**< @brief Waypoints for which the planner succeeded */
+  std::vector<Waypoint::Ptr> failed_waypoints;        /**< @brief Waypoints for which the planner failed */
```

Comment:
Should these be ConstPtr?

Reply:
I don't know. I could imagine a case where you might want to modify the returned waypoints and it would be convenient for them not to be. Perhaps you want to adjust the coefficient values between Descartes and TrajOpt. Or maybe Descartes fails for one waypoint and you want to replan for the succeeded waypoints but adjust some of them to smooth something out.

I'll defer to your judgement.

Reply:
We should a comment that these point to the waypoints provided in the input. So changing them here changes the source. Another alternative is return the index instead of the waypoint itself. 

Also should we update the other planners to fill out the failed and succeed waypoints as part of this PR?

Reply:
Probably. I think I already did descartes, but we could at least return the failed ones from the collision check in trajopt.